onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -24,6 +24,7 @@ OpenVINOExecutionProvider::OpenVINOExecutionProvider(const OpenVINOExecutionProv
   openvino_ep::BackendManager::GetGlobalContext().device_type = info.device_type_;
   openvino_ep::BackendManager::GetGlobalContext().precision_str = info.precision_;
   openvino_ep::BackendManager::GetGlobalContext().enable_vpu_fast_compile = info.enable_vpu_fast_compile_;
+  openvino_ep::BackendManager::GetGlobalContext().num_of_threads = info.num_of_threads_;
```

Comment:
>openvino_ep::BackendManager::GetGlobalContext().num_of_threads = info.num_of_threads_ [](start = 2, length = 85)

maybe here's the best place to validate the value. default it to 8 if invalidate setting encountered like 0, -1.